src/main/java/com/netflix/blitz4j/LoggingContext.java

```diff
@@ -49,6 +56,7 @@
     private ThreadLocal<StackTraceElement> stackLocal = new ThreadLocal<StackTraceElement>();
     private ThreadLocal<LoggingEvent> loggingEvent = new ThreadLocal<LoggingEvent>();
     private ThreadLocal<Level> contextLevel = new ThreadLocal<Level>();
+    private final Set<Category> loggerNeedsLocation = Collections.synchronizedSet(Collections.newSetFromMap(new WeakHashMap<Category, Boolean>()));
```

Comment:
I'll look into a CopyOnWriteArraySet and see what I can do.  As for not caching, when loggers reconfigure I want to be able to start generating location info for them.  If I cache the negative case, I can't.  The negative case is also way cheaper than making a stack trace, so I figured I'm getting a 98% speedup instead of 99.9% speedup.